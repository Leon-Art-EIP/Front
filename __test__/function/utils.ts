import type { ReactTestRendererJSON } from "react-test-renderer";
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from "react-test-renderer";
import { expect } from "vitest";

function toJson(component: renderer.ReactTestRenderer): ReactTestRendererJSON {
  const result = component.toJSON();
  expect(result).toBeDefined();

  return result as ReactTestRendererJSON;
}

export function testSnapshot(element: JSX.Element): void {
  const component = renderer.create(element);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
}
