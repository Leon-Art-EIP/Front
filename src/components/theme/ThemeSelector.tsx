"use client";

import React from "react";
import Themes, { ITheme } from "./Themes";

interface ThemeSelectorProps {
  onSelectColor: (color: string) => void;
  selectedColor: string;
  onThemeSelect: (theme: string) => void; // Ajoutez une fonction pour la sélection de thème
}

const ThemeSelector: React.FC<ThemeSelectorProps> = ({ onSelectColor, selectedColor, onThemeSelect }) => {
  const themeNames = Object.keys(Themes); // Récupérez les noms des thèmes

  const handleColorChange = (color: string) => {
    onSelectColor(color); // Mettre à jour la couleur sélectionnée
  };

  return (
    <div className="p-4 flex items-center">
      {themeNames.map((themeName, index) => {
        const theme: ITheme = Themes[themeName]; // Obtenez le thème correspondant au nom du thème
        return (
          <button
            key={index}
            className="w-10 h-10 mr-2 bg-gray-200 rounded"
            onClick={() => onThemeSelect(themeName)} // Appel de la fonction onThemeSelect avec le nom du thème sélectionné
            style={{ backgroundColor: theme.primary }} // Utilisez une couleur du thème comme couleur de fond du bouton
          ></button>
        );
      })}
    </div>
  );
};

export default ThemeSelector;
