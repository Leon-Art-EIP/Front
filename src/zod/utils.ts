export const getValueOrUndefined = (arg: unknown) =>
  arg === "" || arg === undefined || arg === null ? undefined : arg;
