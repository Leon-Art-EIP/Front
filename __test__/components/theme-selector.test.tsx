import { test } from "vitest";
import { testSnapshot } from "../function/utils";
import ThemeSelector from "../../src/components/theme/ThemeSelector";

test("ThemeSelector snapshot", () => {
  testSnapshot(<ThemeSelector />);
});
