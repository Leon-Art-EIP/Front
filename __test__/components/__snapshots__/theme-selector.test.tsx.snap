// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ThemeSelector snapshot 1`] = `
<div
  className="flex"
>
  <button
    className="rounded-full h-8 w-8 flex items-center justify-center mx-1 focus:outline-none border-gray-400 border-2"
    onClick={[Function]}
    style={
      {
        "backgroundColor": "#ffffff",
      }
    }
  >
    <div
      className="w-4 h-4 rounded-full bg-primary"
      style={
        {
          "marginRight": "4px",
        }
      }
    />
    <div
      className="w-8 h-8 rounded-full bg-tertiary absolute"
    />
  </button>
  <button
    className="rounded-full h-8 w-8 flex items-center justify-center mx-1 focus:outline-none border-gray-400 "
    onClick={[Function]}
    style={
      {
        "backgroundColor": "#333333",
      }
    }
  />
  <button
    className="rounded-full h-8 w-8 flex items-center justify-center mx-1 focus:outline-none border-gray-400 "
    onClick={[Function]}
    style={
      {
        "backgroundColor": "#9c27b0",
      }
    }
  />
  <button
    className="rounded-full h-8 w-8 flex items-center justify-center mx-1 focus:outline-none border-gray-400 "
    onClick={[Function]}
    style={
      {
        "backgroundColor": "#002b36",
      }
    }
  />
  <button
    className="rounded-full h-8 w-8 flex items-center justify-center mx-1 focus:outline-none border-gray-400 "
    onClick={[Function]}
    style={
      {
        "backgroundColor": "#000000",
      }
    }
  />
  <button
    className="rounded-full h-8 w-8 flex items-center justify-center mx-1 focus:outline-none border-gray-400 "
    onClick={[Function]}
    style={
      {
        "backgroundColor": "#f7e1e3",
      }
    }
  />
</div>
`;
